tests/test_reusable_executor.py

```diff
@@ -766,3 +767,35 @@ def leak_some_memory(size=int(1e6), delay=0.01):
         # size + what has leaked since the last memory check.
         for _, leak_size in results:
             assert leak_size / 1e6 < 250
+
+
+def test_reference_cycle_collection():
+    # make the parallel call create a reference cycle and make
+    # a weak reference to be able to track the garbage collected objects
+
+    class A:
+        def __init__(self, size=int(1e6)):
+            self.data = b"\x00" * size
+            self.a = self
+
+    def dummy_func(delay=0.01):
+        from loky import process_executor
+        process_executor._USE_PSUTIL = False
```

Comment:
To reduce the length of the check, maybe you can also monkey-patch `_GC_CHECK_DELAY` to reduce it to `.1` and use delay of `.001`?

Reply:
done.